services/repository/push.go

```diff
@@ -222,7 +222,33 @@ func pushUpdates(optsList []*repo_module.PushUpdateOptions) error {
 				if len(commits.Commits) > setting.UI.FeedMaxCommitNum {
 					commits.Commits = commits.Commits[:setting.UI.FeedMaxCommitNum]
 				}
-				commits.CompareURL = repo.ComposeCompareURL(opts.OldCommitID, opts.NewCommitID)
+
+				oldCommitID := opts.OldCommitID
+				if oldCommitID == git.EmptySHA && len(commits.Commits) > 0 {
+					oldCommit, err := gitRepo.GetCommit(commits.Commits[len(commits.Commits)-1].Sha1)
+					if err != nil {
+						log.Error("gitRepo.GetCommit %s/%s failed: %v", repo.ID, oldCommitID, err)
+					}
+
+					for i := 0; i < oldCommit.ParentCount(); i++ {
```

Comment:
Is it correct to continue to use `oldCommit` if `gitRepo.GetCommit` fails?

Reply:
yup that is gonna NPE.